third_party/java/arcs/build_defs/internal/kotlin.bzl

```diff
@@ -453,14 +453,16 @@ def arcs_kt_plan(name, srcs = [], deps = [], platforms = ["jvm"], visibility = N
             deps = deps,
         )
 
+    deps = [d for d in deps if not d.endswith(".arcs")]
+
```

Comment:
nit: this could be put into a utility method thats shared (e.g. def manifest_deps(deps)) its duped in two places.

Reply:
I like that idea.